packages/http-client-csharp/emitter/src/lib/logger.ts

```diff
@@ -1,20 +1,81 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-import * as winston from "winston";
-
-export enum LoggerLevel {
-  ERROR = "error",
-  WARN = "warn",
-  INFO = "info",
-  DEBUG = "debug",
-  VERBOSE = "verbose",
-}
+import { NoTarget, Program, Tracer } from "@typespec/compiler";
+import { getTracer, reportDiagnostic } from "./lib.js";
+import { LoggerLevel } from "./log-level.js";
+
+export class Logger {
+  private static instance: Logger;
+  private initialized: boolean = false;
+  private tracer: Tracer;
+  private level: LoggerLevel;
+  private program: Program;
+
+  private constructor(program: Program, level: LoggerLevel) {
+    this.tracer = getTracer(program);
+    this.level = level;
+    this.program = program;
+  }
+
+  static initialize(program: Program, level: LoggerLevel): void {
+    if (!Logger.instance) {
+      Logger.instance = new Logger(program, level);
+      Logger.instance.initialized = true;
+    }
+  }
+
+  static getInstance(): Logger {
+    if (!Logger.instance) {
+      throw new Error("Logger is not initialized. Call initialize() first.");
+    }
+    return Logger.instance;
+  }
 
-export const logger: winston.Logger = winston.createLogger({
-  transports: [
-    new winston.transports.Console({
-      level: LoggerLevel.INFO,
-    }),
-  ],
-});
+  info(message: string): void {
+    if (!this.initialized) {
+      throw new Error("Logger is not initialized. Call initialize() first.");
+    }
+    if (
+      this.level === LoggerLevel.INFO ||
+      this.level === LoggerLevel.DEBUG ||
+      this.level === LoggerLevel.VERBOSE
+    ) {
+      this.tracer.trace(LoggerLevel.INFO, message);
+    }
+  }
+
+  debug(message: string): void {
+    if (!this.initialized) {
+      throw new Error("Logger is not initialized. Call initialize() first.");
+    }
+    if (this.level === LoggerLevel.DEBUG || this.level === LoggerLevel.VERBOSE) {
+      this.tracer.trace(LoggerLevel.DEBUG, message);
+    }
+  }
+
+  verbose(message: string): void {
+    if (!this.initialized) {
+      throw new Error("Logger is not initialized. Call initialize() first.");
+    }
+    if (this.level === LoggerLevel.VERBOSE) {
+      this.tracer.trace(LoggerLevel.VERBOSE, message);
+    }
+  }
+
+  warn(message: string): void {
+    reportDiagnostic(this.program, {
+      code: "General-Warning",
+      format: { message: message },
+      target: NoTarget,
+    });
+  }
+
+  error(message: string): void {
```

Comment:
woudl also be great that those methods are not used and instead use the `reportDiagnositc` with explicitly defined errrors/warning as much as possible

Reply:
Agreed, this gives us at least a way to surface our current errors and warnings but we will be migrating them to specific diagnostics over time

Reply:
you could put  that on it to get ide note 
```
/**
 * @deprecated use reportDiagnostic
 */
```

Reply:
Adding this follow up item to track https://github.com/microsoft/typespec/pull/3343#discussion_r1599202513